packages/db/src/runtime/db-client.ts

```diff
@@ -53,17 +53,38 @@ export function createRemoteDatabaseClient(options: RemoteDbClientOptions) {
 
 	return options.dbType === 'studio'
 		? createStudioDatabaseClient(options.appToken, remoteUrl)
-		: createRemoteLibSQLClient(options.appToken, remoteUrl);
+		: createRemoteLibSQLClient(options.appToken, remoteUrl, options.remoteUrl.toString());
 }
 
-function createRemoteLibSQLClient(appToken: string, remoteDbURL: URL) {
+function createRemoteLibSQLClient(appToken: string, remoteDbURL: URL, rawUrl: string) {
 	const options: Partial<LibSQLConfig> = Object.fromEntries(remoteDbURL.searchParams.entries());
 	remoteDbURL.search = '';
 
+	let url = remoteDbURL.toString();
+	if (remoteDbURL.protocol === 'memory:') {
+		// libSQL expects a special string in place of a URL
+		// for in-memory DBs.
+		url = ':memory:';
+	} else if (
+		remoteDbURL.protocol === 'file:' &&
+		remoteDbURL.pathname.startsWith('/') &&
+		!rawUrl.startsWith('file:/')
+	) {
+		// libSQL accepts relative and absolute file URLs
```

Comment:
Do you think it's worth opening an issue in their repository? This feels like a bug to me

Reply:
I don't think so. URLs have no way to differentiate a relative path and an absolute path. All of these are parsed to the exact same value by `new URL`
- `file:foo.db`
- `file:./foo.db`
- `file:/foo.db`
- `file:///foo.db`


So they had to not comply with URL in order to support both relative and absolute paths. It's more of a feature than a bug. A feature with unfortunate consequences.

Reply:
I'll bring this up on the libsql-js Discord channel to confirm

Reply:
libSQL team is deciding what to make of this. We can merge our workaround to unblock people until some other option is decided.

Reply:
Agree. Let's merge it